### DISCUSSION TITLE
Header Menus

### DISCUSSION THREAD

**Discussion by giscus** [hbstack/site#430] [General]
# Header Menus

This article describes how to configure the header&rsquo;s menus.

https://hbstack.dev/en/modules/header/menus/

**Comment by bigkevuk** [comment]
HI Razon, I can't see where in the parameters how I can make the menu go along time as per the screenshot here. 
It's always the ... as per the site is currently displaying this page if that makes sense.

So to be clear I have done a clean install and the menu is under the ... at the top right. Just like it is for this website.
What I would like to do is to have the items across the top as per the image on the top of this page.

Hope this makes sense and any help apprecaited.

**Reply by razonyang** [reply]
Could you please provide screenshots what you want to implement? I don't get it.

**Reply by razonyang** [reply]
If you mean the social links (icon buttons) on the right, see https://hbstack.dev/en/modules/header/socials/.

![image](https://github.com/hbstack/site/assets/17720932/be6fddc5-6160-4aa9-9f5e-633e87d3d79a)

If you mean the header image/background, see https://hbstack.dev/en/modules/header/banner/.

> The banner is not included in the theme, you should import the module yourself and tweak the parameters.

![image](https://github.com/hbstack/site/assets/17720932/afad17a8-13dc-49e2-a054-bbc2c854bcfc)

If you're requesting a new feature, please file an issue for it and describe it in detail if possible.

**Reply by razonyang** [reply]
The header parts as follows.

![sss](https://github.com/hbstack/site/assets/17720932/b7283bfd-67f6-45c0-929f-11602f0059a8)

You can find the working configurations and examples on the docs and original theme template (github.com/hbstack/theme).
Please feel free to ask if you don't remember and can't find where it is.

**Comment by bigkevuk** [comment]
I think the media queries are set-up wrong and see where the confusion is from my side :-) 

**1920x1080 - Desktop View**
![image](https://github.com/hbstack/site/assets/978928/a51aa4fd-e55c-4ed3-91db-8f192894fbb2)
**1600x900 - Desktop View**
![image](https://github.com/hbstack/site/assets/978928/18579eb3-f7c7-4bb4-b20b-f6cb2062481b)
**1440x900 - Desktop View**
![image](https://github.com/hbstack/site/assets/978928/01d590aa-a7c2-4ffc-b6ae-a32b9d0f563d)

**1365x768 - Mobile View**
![image](https://github.com/hbstack/site/assets/978928/98eec8b3-9de3-4e49-9dfb-8d8e02a99581)
**1280x960 - Mobile View**
![image](https://github.com/hbstack/site/assets/978928/1830f038-6fb1-4e12-a388-901f7ce6f3f6)

Lenevo X13 Yoga Screen: 2560x1600 @ 200% recommended scale give **Mobile View**
![image](https://github.com/hbstack/site/assets/978928/5bc2b2fb-5a1b-4d04-abd1-21b4217b563f)
Changing to 100% makes it a a lot harder to use but does give **desktop view**
![image](https://github.com/hbstack/site/assets/978928/fad64b67-fda6-4cec-a6b7-ef7962de3ddb)

**Reply by razonyang** [reply]
The theme is designed to collapse the menus when the width less than 1400px, and expand when the width is great than that point.
Please file a new issue for this if you want to see it can be configurable, but it may not happen so fast, I think there are some stuff need to be pay attention.

**Comment by bigkevuk** [comment]
Thanks for the heads up :-) will take a look

**Comment by TetraTheta** [comment]
How can I use custom SVG/PNG for icon of menu? [Icon module](https://hugomods.com/en/docs/icons/) doesn't seem to support 'custom icon'.

For example, I'd like to add icon as transparent PNG or custom SVG which is not provided from venders of Icon module.  
Since subject of my custom icon is not popular, I can't ask those vendors to add my icons.

![250](https://github.com/hbstack/site/assets/37061801/0326ec10-737d-4566-b782-4fd5b9fe5da4)  
↑ Example of 'custom icon'.

Currently, it seems without forking or overrideing Menus module, I can't use my icons in menu.

---

Also, how can I remove 'grey gradiented rounded square backgroud' from menu icon? I want it to have different style or not displayed.

![IMG1695368164](https://github.com/hbstack/site/assets/37061801/f00baa9e-8d87-4421-9cbc-cca9d64aa4fb)  
↑ This 'grey gradiented rounded square backgroud'.

**Reply by razonyang** [reply]
You can just put your SVG (not PNG) icons on `assets/icons/whatever` (`whatever` is the vendor name, replace with whatever you want), and then you are able to use it in menus.

You can override the via [custom styles](https://hbstack.dev/en/docs/advanced/style/) with the `!important` option. I don't suggest to override the layout, since there is no gurantee there won't be BREAKING CHANGES on the layouts/partials.

```css
.dropdown-item-icon {
    background-color: transparent !important;
}
```

**Reply by TetraTheta** [reply]
If then, how can I use PNG icon? Like I said, I want to use PNG icon because I can't find SVG icon that I can use.

**Reply by razonyang** [reply]
You can either 1) specify the raw PNG `<img>` HTML as the [`pre` or `post` property](https://gohugo.io/content-management/menus/#properties-front-matter), or 2) convert the PNG to SVG and follow the steps I mentioned above.

**Reply by TetraTheta** [reply]
Thank you!

I've managed to convert PNG to SVG, so I can use it without making HTML code stuff.
